cmd/sink/filters/filters.go

```diff
@@ -23,12 +23,18 @@ import (
 )
 
 const (
-	globalKey       = "kubearchive"
+	globalKeyEnvVar = "KUBEARCHIVE_NAMESPACE"
 	mountPathEnvVar = "MOUNT_PATH"
 )
 
 var ErrNoGlobal = errors.New("no global expressions exist")
 
+// getGlobalKey returns the key where global filters can be found in the sink-filters ConfigMap. Will return empty
+// string if the environment variable with the name specified by globalKeyEnvVar is not set.
+func getGlobalKey() string {
```

Comment:
What is the advantage of having a function in this case?

Reply:
I agree that calling `os.Getenv(globalKeyEnvVar)` twice could be avoided.

Let me suggest the usage of an IIF ([Immediately Invoked Function](https://golangbyexample.com/function-golang-complete-guide/#IIF_or_Immediately_Invoked_Function))
```go
var globalKey = func() string {
    return os.Getenv(globalKeyEnvVar)
}()
```

Reply:
> Let me suggest the usage of an IIF ([Immediately Invoked Function](https://golangbyexample.com/function-golang-complete-guide/#IIF_or_Immediately_Invoked_Function))

But I still don't see then point of a one line IIF.  Why not just call `os.Getenv` directly?

Reply:
maybe in the future we decide that `globalKey` won't get passed as an environment variable. we could just change this function instead of having to change it in 4 different places

Reply:
also, I know that `os.Getenv` isn't a instruction that has a high cost, but I feel that it's ok to load this parameter just once. In general terms I like to prevent doing reads, even from the environment variables, if it's not needed.

Reply:
turns out that with Marta's suggestion as is, it the IIF can be called more than once. I moved the call to `os.Getenv` to the `init` function which Go guarantees will only be run once even if the module is imported multiple times

Reply:
interesting, I thought that it would run only once